shell/platform/fuchsia/flutter/flatland_connection.cc

```diff
@@ -67,13 +71,19 @@ void FlatlandConnection::DoPresent() {
 
 // This method is called from the UI thread.
 void FlatlandConnection::AwaitVsync(FireCallbackCallback callback) {
-  if (first_call_to_await_vsync_) {
+  // Immediately fire callbacks until the first Present. We might receive
+  // multiple requests for AwaitVsync() until the first Present, which relies on
+  // receiving size on FlatlandPlatformView::OnGetLayout() at an uncertain time.
+  bool first_present_called;
```

Comment:
Is acquiring and releasing this lock a second time on every frame expensive? It seems like we're adding additional work to every frame to handle a case that's only true for the first couple of vsyncs and never afterwards.

If this is cheap ignore this comment.

Reply:
Good call! I am changing so that we keep the lock. I was trying to handle the case that FlatlandConnectionTest.PresentCreditExhaustion exercises where FireCallbackCallback runs Present(). But it is not applicable in production code as we have a thread jump. AwaitVsync is always on UI thread and Present() is on raster thread. I am modifying the test.